test/utils_request.js

```diff
@@ -104,308 +53,111 @@ describe('Apify.utils_request', () => {
-                .to
-                .eql(CONTENT);
-        });
-
-        it('decompress brotli', async () => {
-            const options = {
-                url: `http://${HOST}:${port}/brotli`,
-
-            };
-
-            const response = await requestExtended(options);
-            expect(response.body).to.eql(CONTENT);
-        });
-
-        it('it does not throw error for 400+ error codes when throwOnHttpError is false', async () => {
-            const options = {
-                url: `http://${HOST}:${port}/500`,
-
-            };
-            let error;
-            try {
-                await requestExtended(options);
-            } catch (e) {
-                error = e;
-            }
-            expect(error).to.be.undefined; // eslint-disable-line
-        });
-
-        it('it does not throw error for 400+ error codes when throwOnHttpError is true', async () => {
-            const options = {
-                url: `http://${HOST}:${port}/500`,
-                throwOnHttpError: true,
-
-            };
-            let error;
-            try {
-                await requestExtended(options);
-            } catch (e) {
-                error = e;
-            }
-            expect(error.message).to.exist; // eslint-disable-line
-            expect(error.message.includes(ERROR_BODY)).to.be.eql(true);
-        });
-
-        it('it throws error when the body cannot be parsed and the code is 500 when throwOnHttpError is true', async () => {
-            const options = {
-                url: `http://${HOST}:${port}/500/invalidBody`,
-                throwOnHttpError: true,
-
-            };
-            let error;
-            try {
-                await requestExtended(options);
-            } catch (e) {
-                error = e;
-            }
-            expect(error.message).to.exist; // eslint-disable-line
-        });
-
-        it('it throws error when the body cannot be parsed', async () => {
-            const options = {
-                url: `http://${HOST}:${port}/invalidBody`,
-
-            };
-            let error;
-            try {
-                await requestExtended(options);
-            } catch (e) {
-                error = e;
-            }
-            expect(error.message).to.exist; // eslint-disable-line
-        });
-
-        it('it returns json when 500 even if content-type is different, throwOnHttpError is true ', async () => {
-            const options = {
-                url: `http://${HOST}:${port}/500/json`,
-                throwOnHttpError: true,
-
-            };
-            let error;
-            try {
-                await requestExtended(options);
-            } catch (e) {
-                error = e;
-            }
-            expect(error.message).to.exist; // eslint-disable-line
-            expect(error.message.includes(JSON_BODY.message)).to.be.eql(true);
-        });
-    });
-
     describe('Apify.requestAsBrowser', async () => {
```

Comment:
Can we also test that the order of headers is same as in Firefox? I think that's quite important.